### PR TITLE
Simplify Bisector

### PR DESCRIPTION
**Pull request by thetheodor**

### CODE REVIEW

File: diopter/bisector.py [hunk 1]
```diff
@@ -1,258 +1,165 @@
...
-        # Get size of range
-        range_size = len(repo.direct_first_parent_path(midpoint, bad_rev))
-
-        # Move 10% towards the last bad
-        step = max(int(0.9 * range_size), 1)
-        midpoint = repo.rev_to_commit(f"{bad_rev}~{step}")
+from diopter.utils import run_cmd
+
+GitRevision: TypeAlias = str
+
```
Comment (Quarub) [inline diopter/bisector.py:10]: Do you want to differentiate between a revision and a commit hash?
The commit hash is 'pure' and (mostly) won't change while a revision (`main`) potentially can.

Comment (thetheodor) [reply diopter/bisector.py]: That was stale. I forgot to delete it

Comment (Quarub) [reply diopter/bisector.py]: Yeah that's something I never did and I cost me a confusing debug session once.

**Comment by Quarub** [comment]
Do you know how `git bisect` behaves when it encounters a merged commit? 
Say the history `A - B - BRANCH -C  - MERGE - D`. Also assume C is good and MERGE is bad.
Will it  go bisect in the branch that lead to the merge commit?
Similar question for cherry picked commits. I once ended up in an obscure branch of BOLT when I did not specify following the `first-parent`. :sweat_smile:

**Pull request merged**